features/step_definitions/yamlmodification.rb

```diff
@@ -0,0 +1,5 @@
+Given /^I have the machine yaml file (.+) and I add the taints (.+)$/ do |filename,taints|
+  struct = YAML.load_file(localhost.absolutize(filename))
+  struct["spec"]["taints"]=taints
```

Comment:
if this fails with `undefined method `[]' for false:FalseClass` then perhaps `machines_to_taint.yml` does not have a `spec` element. I don't see that file content here.

Reply:
I couldnt figure out why the spec is not populating because , when I ran the same as one steps after the other in pry is was having data